tests/test_files.py

```diff
@@ -68,6 +73,76 @@ def test_11_files_count(self):
         self.assertIn('files', data)
         self.assertEqual(data['files'], 1)
 
+    def test_12_files_keys(self):
+        """Test the 'keys' and all-keys' arguments."""
+        self.start_server()
+        token = self.get_token()
+        r = RestClient(self.address, token, timeout=1, retries=1)
+
+        metadata = {
+            "logical_name": "blah",
+            "checksum": {"sha512": hex("foo bar")},
+            "file_size": 1,
+            "locations": [{"site": "test", "path": "blah.dat"}],
+            "extra": "foo",
+            "supplemental": ["green", "eggs", "ham"],
+        }
+        data = r.request_seq("POST", "/api/files", metadata)
+        self.assertIn("_links", data)
+        self.assertIn("self", data["_links"])
+        self.assertIn("file", data)
+        assert "extra" not in data
+        assert "supplemental" not in data
+        url = data["file"]
+        uid = url.split("/")[-1]
+
+        # w/o all-keys
+        data = r.request_seq("GET", "/api/files")
+        assert set(data["files"][0].keys()) == {"logical_name", "uuid"}
+
+        # w/ all-keys
+        body = {"all-keys": True}
+        data = r.request_seq("GET", "/api/files", body)
```

Comment:
GET Request + Request Body are like ketchup and vanilla ice-cream, individually great but don't belong together.
https://stackoverflow.com/a/983458

I know this is just a test case, and I actually like the clean look; it's easy to tell which query parameters you're setting (and what their values are) without mentally parsing the query string for a question mark.

I'd be happy with a comment that said "`# body shouldn't be sent to a GET request, but we're only testing so we'll give ourselves a pass`" or you can rewrite the query strings.

Reply:
While it looks like [Roy agrees with you](https://stackoverflow.com/a/983458/13156561), our ice cream has been ketchup-y since the beginning: https://github.com/WIPACrepo/file_catalog/blame/64dc55510308a0fa98145b3d1c2d7628bbbb4fcf/file_catalog/server.py#L384

Reply:
Yeah, I don't think you need to rewrite `request_seq` or anything deep here.

I'm just suggesting to add a comment to the test to show that it's not recommended practice; it's just a test, not production code, so a little cheating is fine in this context.

Or, rewrite the test to put the parameters into the query string and remove body altogether. This second option requires more work and (IMHO) obscures the purpose of the test, so I think a comment is better.

Trying to fix it so no GET requests can possibly have a request body is way beyond the scope of this PR; please don't do that here. :smile:

Reply:
I think there's some confusion here about how the rest client works.  The third function parameter is for arguments, and is not directly a message body.  If it is a GET or HEAD request, they get converted to query parameters.  Otherwise, they are sent as json in the body.

So, I think the only thing needed here is renaming the variable from `body` to something like `args`.

Reply:
Ahhh... that works too!

Reply:
Thanks for the clarification @dsschult. I've always thought about that argument as the "body" of the _rest_tools_ request, and _not_ the `body` of the REST request. At any rate, I renamed the variable.